lib/src/Uncertainty/Bayesian/IndependentMetropolisHastings.cxx

```diff
@@ -0,0 +1,130 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "openturns/IndependentMetropolisHastings.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+
+CLASSNAMEINIT(IndependentMetropolisHastings)
+
+static const Factory<IndependentMetropolisHastings> Factory_IndependentMetropolisHastings;
+
+/* Default constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings()
+  : MetropolisHastingsImplementation()
+{
+  // Nothing to do
+}
+
+
+/* Parameters constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings(const Distribution & targetDistribution,
+                                                          const Point & initialState,
+                                                          const Distribution & proposal,
+                                                           const Indices & marginalIndices)
+  : MetropolisHastingsImplementation(targetDistribution, initialState, marginalIndices)
+{
+  setProposal(proposal);
+}
+
+/* Parameters constructor */
+IndependentMetropolisHastings::IndependentMetropolisHastings(const Function & targetLogPDF,
+                                                          const Domain & support,
+                                                          const Point & initialState,
+                                                          const Distribution & proposal,
+                                                           const Indices & marginalIndices)
+  : MetropolisHastingsImplementation(targetLogPDF, support, initialState, marginalIndices)
+{
+  setProposal(proposal);
+}
+
+
+/* String converter */
+String IndependentMetropolisHastings::__repr__() const
+{
+  return OSS() << "class=" << IndependentMetropolisHastings::GetClassName()
+         << " name=" << getName()
+         << " derived from " << MetropolisHastingsImplementation::__repr__()
+         << " proposal=" << proposal_;
+}
+
+
+IndependentMetropolisHastings* IndependentMetropolisHastings::clone() const
+{
+  return new IndependentMetropolisHastings(*this);
+}
+
+
+Point IndependentMetropolisHastings::getCandidate() const
+{
+  const Point candidate(proposal_.getRealization());
+
+  // LogPDF of the proposal distribution at the candidate point
+  const Scalar logProbNew = proposal_.computeLogPDF(candidate);
+
+  // LogPDF of the proposal distribution at the current point
+  const UnsignedInteger proposalDimension = proposal_.getDimension();
+  Point current(proposalDimension);
+  for (UnsignedInteger j = 0; j < proposalDimension; ++ j)
+  {
+    current[j] = currentState_[marginalIndices_[j]];
+  }
+  const Scalar logProbCurrent = proposal_.computeLogPDF(current);
```

Comment:
you can simplify and use select now: ```Point current(currentState_.select(marginalIndices_));```

Reply:
Yes, indeed! Can you please apply my GitHub suggestion @jschueller ? I spoke with Merlin, and he is fine with it.